### PR TITLE
Improve User Management Documentation

### PR DESCRIPTION
**Pull request by egiurleo**
- Remove user documentation from the Authentication tutorial
- Create a separate User Management tutorial with information about creating, viewing, updating, and removing users

### CODE REVIEW

File: docs/tutorials/ruby-driver-user-management.txt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The Mongo Ruby Driver provides a set of methods for managing users on a MongoDB deployment.
```
Comment (p-mongo) [inline docs/tutorials/ruby-driver-user-management.txt]: on -> in ?

File: docs/tutorials/ruby-driver-user-management.txt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+when performed on a replica set.
+
+.. seealso::
+  :manual:`Built-in roles</reference/built-in-roles/>`
+  :manual:`Write Concerns</core/replica-set-write-concern/>`,
+  :ref:`Sessions<sessions>`,
+
+User Info
+`````````
+To view information about a user that has already been created in the database, use the ``info`` method:
```
Comment (p-mongo) [inline docs/tutorials/ruby-driver-user-management.txt]: has already been created -> already exists ?

File: docs/tutorials/ruby-driver-user-management.txt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+roles. If the user doesn't exist, this method will return an
+empty Array.
+
+The ``info`` method also takes an optional ``Hash`` of options
+as a second argument. Currently, the only supported option is ``:session``,
+which allows you to specify a ``Mongo::Session`` object to use for this
+operation.
+
+The Ruby Driver does not have a method that lists all of the users
+that currently exist on a database.
```
Comment (p-mongo) [inline docs/tutorials/ruby-driver-user-management.txt]: on -> in ?

File: docs/tutorials/ruby-driver-user-management.txt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+  client.database.users.remove('alanturing')
+
+You may pass a ``Hash`` of options as a second argument. The two supported
+options for the ``remove`` method are ``:session`` and ``:write_concern``.
+``:session`` allows you to specify a ``Mongo::Session`` object to use for
+this operation. ``:write_concern`` specifies the write concern
+of the operation if you are running this command against a replica set.
+
+The Ruby Driver does not implement a method for removing all users
```
Comment (p-mongo) [inline docs/tutorials/ruby-driver-user-management.txt]: implement -> provide ?

File: docs/index.txt [hunk 5]
```diff
@@ -62,6 +62,7 @@ For tutorials on Mongoid, see the `Mongoid Manual <https://docs.mongodb.com/mong
       quick-start
       /tutorials/ruby-driver-create-client
       /tutorials/ruby-driver-authentication
+      /tutorials/ruby-driver-user-management
```
Comment (p-mongo) [inline docs/index.txt]: Could this be renamed to just `/tutorials/user-management`?

Comment (p-mongo) [reply docs/index.txt]: I'd like to eventually get rid of all of these ruby-driver prefixes.

File: docs/tutorials/user-management.txt [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    roles: [ Mongo::Auth::Roles::READWRITE ]
+  })
+
+  client.database.users.create(user)
+
+The ``create`` method takes a ``Hash`` of options as an optional second argument.
+The ``:roles`` option allows you to grant permissions to the new user.
+For example, the ``Mongo::Auth::Roles::READ_WRITE`` role grants the user the ability to both
+read from and write to the database in which they were created. Each role can be specified
+as a ``String`` or as a ``Hash``. If you would like to grant permissions to a user on a database
```
Comment (p-mongo) [inline docs/tutorials/user-management.txt]: I would love also for there to be a clarification (not necessarily made as part of this PR) as to what the auth source is for the created user. I.e. if I create a user on `foo` database's view, does the user have rights to `foo` database with an auth source of `admin` or is the user created with auth source of `foo`?

Comment (egiurleo) [reply docs/tutorials/user-management.txt]: Oh that's a good point. I'm still waiting to get the AWS access I need to deploy, so I'll just add it to this PR while I wait.

Comment (egiurleo) [reply docs/tutorials/user-management.txt]: I lied. I'll merge this and write up some information about auth source in a follow-up PR.

**Comment by p-mongo** [comment]
Ticket: https://jira.mongodb.org/browse/RUBY-1746

**Pull request merged**